File: 12 Integer to Roman/Java/Solution.java
class Solution {
    public String intToRoman(int num) {
        int[] integers = new int[]{1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        String[] symbols = new String[]{"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};

        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < integers.length; i++) {
            if (num == 0) {
                return sb.toString();
            }
            int integer = integers[i];
            int n = num / integer;
            for (int j = 0; j < n; j++) {
                sb.append(symbols[i]);
            }
            num %= integer;
        }

        return sb.toString();
    }
}

File: 12 Integer to Roman/Python/Solution.py
from typing import List, Self


class Solution:
    def intToRoman(self: Self, num: int) -> str:
        integers: List[int] = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
        symbols: List[str] = ['M', "CM", 'D', "CD", 'C', "XC", 'L', "XL", 'X', "IX", 'V', "IV", 'I']

        res: str = ''
        i: int = 0
        while i < len(integers):
            if num == 0:
                return res
            n: int = num // integers[i]
            res += n * symbols[i]
            num %= integers[i]
            i += 1
        
        return res 